### DISCUSSION TITLE
Database images

### DISCUSSION THREAD

**Discussion by HacheEle** [jolpica/jolpica-f1#91] [Q&A] [answered]
Hello:
Is it possible to download the database as data dump? (as it could be done on http://ergast.com/mrd/db/)
Thank you in advance

**Comment by jolpica** [comment, marked as answer]
Currently we don't provide any database dumps.
While in alpha we actually retrieve our data from those ergast Database images, so for now we'd suggest downloading from there instead.

We will revisit this next season when we are using our own data, but it will likely be in a different format to Ergast.

**Reply by Vucko95** [reply]
Once new season arrives, will all the data be restructured to the new format, once Ergast is EoL ?

**Reply by jolpica** [reply]
Once the new season arrives, the existing API endpoints will continue to function as expected.

Behind the scenes, we are already storing the data in a new format, which will hopefully increase the quality of data in the long run. This is why database dumps will very likely be in a different format, if we provide them.

**Comment by awanderingspirit** [comment]
This has to be the single most important issue with this API. 

Notice that...

1. no database dumps are provided (i.e. plain files one can repeatedly access offline), 
2. rate limits for requests are put in place,
3. and offset (of rows) is set to a max of 100.

Ergast empowered users with the kind provision of plain old csv files that are programming language-agnostic. Doesn't this measure address the challenge of facing insurmountable amounts of HTTP requests by leaving a large portion of users with promptly accessible offline images of the database?

One has to write an elaborate script in order to convert the JSON endpoints of this database to reusable offline files. The task is challenging given that you would encounter endless fields leading to more irreducible records (denoted in the database diagram as  _foreign keys_, see: [picture](https://[raw.githubusercontent.com/jolpica/jolpica-f1/63220e020d687291202ab1032a5dc6a17f58c42b/docs/database.svg](https://raw.githubusercontent.com/jolpica/jolpica-f1/63220e020d687291202ab1032a5dc6a17f58c42b/docs/database.svg))), not to mention points 2 and 3 above.

**Reply by Oldertarl** [reply]
I honestly do not see where the issue is when not running the postgres DB locally as well defined in the github definitions.

I have written a simple php script to validate if the json data (file) is present on my webserver to download the data, if not - call the API and save the data for future use.
When there is a offset, I iterate through the pages with a small delay to not hit the throttle limit.
This is faster than downloading the API calls every time they are needed.
Most of the data is static and if an update is provided on data, I simply delete the json file locally to have it downloaded again.

**Reply by jolpica** [reply]
Thanks for the feedback.

All work done on this API is done by a small team in our free time, unfortunately this means we don't always have time to do everything we'd like to.

We do plan to release database images in the future, but at the moment our focus is on ensuring we can import data in a reliable manner.

To set expectations, when we do eventually release database dumps, they will *not* be in an ergast compatible format, they will likely follow our own DB schema which we have a diagram of in the read me.

**Reply by gtibrett** [reply]
@jolpica Where are you importing the data from? i.e. for 2025 and on, what is the original source of data being imported into the DB? Ergast? some other source?

I would be happy to take on the task of turning that imported data into csv/sql

**Reply by harningle** [reply]
@gtibrett it's going to be PDFs in fia website, and we are working on the data import now: https://github.com/harningle/fia-doc

**Comment by ErwinBramer** [comment]
First of all, I want to thank you for your wonderful work with this API. 
Answering the original question, I am working on a solution in R to update the Ergast .csv files with new data from the Jolpica F1 API. I think I'm on 90% now. I started this for my own use, but I think there are a lot of people looking for updated .csv files. I also noticed that the much referenced contribution on Kaggle by [Vopani](https://www.kaggle.com/datasets/rohanrao/formula-1-world-championship-1950-2020) is being discontinued. 
As said, I started this project updating the Ergast database for my own use, but now I am thinking about publishing it on Kaggle. Before I will do this, I want to be sure this is OK for you @jolpica.

**Reply by jolpica** [reply]
I'd ask to please refrain from polling the API to update database dumps. However I understand I haven't been helpful in providing an alternative. I'll post a sample below of a database dump we'll hope to provide in the future.
Ideally please make use of this, or use it as a basis to convert into the old ergast format yourself. I'd highly recommend using the new format though, as we aim to start providing more data than the original ergast had available.

**Reply by ErwinBramer** [reply]
Thanks for your quick reply. It would be wonderful if you could provide something like the alpha dump below. In that case I wouldn't need to poll your API. Since I am relatively new in data science, I know little to nothing of databases. My knowledge is is limited to .csv files and data files R can read. That's at least the way I used the old Ergast data.

**Comment by jolpica** [comment]
Please understand that the following is an **alpha** and is subject to change, whether that be file names, column names or anything else.
This dump is also currently out of date with production data (2025 data), but it will hopefully serve as a basis to help prepare for when we release these in the future.

[ALPHA SUBJECT TO CHANGE DUMP](https://github.com/user-attachments/files/18815806/dump.zip).

**Reply by jonas-lundqvist** [reply]
Not to put pressure on you in any way, but is there a new db dump coming soon that would include any 2025 data?

**Reply by jolpica** [reply]
Check our Ko-fi page, theres a dump up to 2025 round 2 (it's free).

There's unlikely to be anymore dumps until either  we automate the dumps or the end of the season (whichever comes first)

**Reply by Vucko95** [reply]
Has there been any progress on automating the database dumps? I'm currently relying exclusively on the latest Ergast DB dump for my website to avoid excessive public source fetching, and I'd be happy to assist if any help is needed with automation efforts.

**Reply by jonas-lundqvist** [reply]
> Has there been any progress on automating the database dumps? I'm currently relying exclusively on the latest Ergast DB dump for my website to avoid excessive public source fetching, and I'd be happy to assist if any help is needed with automation efforts.

I just noticed this from five days ago 👀
https://github.com/jolpica/jolpica-f1/pull/253

**Comment by jolpica** [comment]
Heres a CSV database dump from today, we're very close to getting these automated.
[csv_dump_2025-08-31_81.zip](https://github.com/user-attachments/files/22067756/csv_dump_2025-08-31_81.zip)

To set expectations, we hope to use dumps like this to help fund the API and encourage other projects to donate to our costs if they want to use our latest data.
The current plan is to provide free dumps delayed by a rolling timeframe, or to just provide season end/mid-season dumps for free, then have the latest dumps available for contributors or monetary supporters. 

Any feedback is welcomed, and I hope to get these out weekly soon.

**Comment by jolpica** [comment]
Dumps are now available, see this announcement
#261

**Reply by HacheEle** [reply]
As the one who initiated this request, I want to sincerely thank you for addressing it.
Your response is greatly appreciated and has been very helpful.
